test/e2e/e2e_test.go

```diff
@@ -207,10 +220,82 @@ func requireReplicatedStream(t *testing.T, c1 *catalystContainer, c2 *catalystCo
 		content := string(body)
 		for _, expected := range []string{"RESOLUTION=1920x1080", "FRAME-RATE=30", "index.m3u8"} {
 			if !strings.Contains(content, expected) {
+				glog.Info("Failed to get HLS manifest")
 				return false
 			}
 		}
+		glog.Info("Got HLS manifest!")
 		return true
 	}
 	require.Eventually(t, correctStream, 5*time.Minute, time.Second)
+
+	// Define a set of protocols to test at output of node catalyst-two
+	protocols := map[string]string{
```

Comment:
Nice! I'm thinking that each protocol should probably be its own test, rather than just having one big loop in here? Minimally, having a separate function for running a protocol test will let you `return err` in the error cases for those instead of having to do awkward `continue`s or `break`s

Reply:
Right, I think I got a little lost in trying to do things in the "idiomatic" go style. I'll take a look at refactoring.

Reply:
Yeah, it should be a separate test. Also, the very idiomatic in Go is to use a tests [] array. Please check "Introducing table driven tests" in https://dave.cheney.net/2019/05/07/prefer-table-driven-tests